src/cppmangle.c

```diff
@@ -164,7 +164,8 @@ class CppMangleVisitor : public Visitor
                         else
                         {
                             dinteger_t val = e->toInteger();
-                            if (val < 0)
+                            // TODO: try and do this without knowing what dinteger_t type is
+                            if ((long)val < 0)
```

Comment:
Ah ok I got it. I admit a lot of ignorance in how to add tests for dmd. I will look into it. I was just fixing the warning complaint :)